source/routing/raptor.cpp

```diff
@@ -248,9 +248,11 @@ static void init_best_pts_snd_pass(const RAPTOR::vec_stop_point_duration& depart
                                    IdxMap<type::StopPoint, DateTime>& best_labels) {
     for (const auto& d: departures) {
         if (clockwise) {
-            best_labels[d.first] = departure_datetime + d.second.total_seconds() - 1;
+            best_labels[d.first] = std::min(best_labels[d.first],
```

Comment:
yes, and that's almost checked by the type system! See utils/idx_map.h for more information.